domains/auth/Auth.API/API/Repository/TokenStorage.cs

```diff
@@ -11,5 +13,40 @@ public string GetIdTokenByOpaqueToken(string token)
         {
             throw new NotImplementedException();
         }
+
+        public bool InternalTokenValidation(InternalToken interalToken)
+        {
+            if (interalToken == null)
+            {
+                return false;
+            }
+
+            if (interalToken.Issued > DateTime.UtcNow || interalToken.Expires < DateTime.UtcNow)
+            {
+                return false;
+            }
+
+            return true;
+        }
+
+        public InternalToken? GetInteralTokenByOpaqueToken(string token)
```

Comment:
@C-Christiansen I might have a suggestion to this method. Lets have a talk tomorrow 👍